onadata/libs/utils/logger_tools.py

```diff
@@ -127,29 +127,48 @@ def get_uuid_from_submission(xml):
     return len(split_xml) > 1 and split_xml[1] or None
 
 
-def get_xform_from_submission(xml, username, uuid=None):
-    # check alternative form submission ids
-    uuid = uuid or get_uuid_from_submission(xml)
-
-    if not username and not uuid:
+def get_xform_from_submission(
+        xml, request, username, uuid=None):
```

Comment:
I think I am not redefining existing functions by introducing new parameters before existing parameters. I would perhaps go by adding a new function or adding the new parameter as optional. And perhaps work towards a deprecation process.
Can we leave it at?
```python
def get_xform_from_submission(xml, username, uuid=None, request=None):
...
if request:
   # new way
else:
   # existing old way.
```

Reply:
Sure, I'll make this change.